packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -71,12 +106,266 @@ export class IamPolicyChecksWebview extends VueWebview {
+                'yellow',
+            ])
+            void vscode.commands.executeCommand('workbench.actions.view.problems')
+        }
+    }
+
+    public executeCustomPolicyChecksCommand(command: string) {
+        try {
+            const response = execSync(command)
+            this.handleCustomPolicyChecksCliResponse(response.toString())
+        } catch (err: any) {
+            if (err.status === 1) {
+                this.onCustomPolicyCheckResponse.fire([getCliErrorMessage(err.message), 'red'])
+            } else {
+                this.handleCustomPolicyChecksCliResponse(err.stdout.toString())
+            }
+        }
+    }
+
+    public handleCustomPolicyChecksCliResponse(response: string) {
+        const diagnostics: vscode.Diagnostic[] = []
+        const jsonOutput = JSON.parse(response)
+        if (jsonOutput.BlockingFindings.length === 0 && jsonOutput.NonBlockingFindings.length === 0) {
+            this.onCustomPolicyCheckResponse.fire([
+                'Result: PASS. Policy checks did not discover any problems with your policy.',
+                'green',
+            ])
+        } else {
+            jsonOutput.BlockingFindings.forEach((finding: any) => {
+                finding.details.reasons.forEach((reason: any) => {
+                    diagnostics.push(
+                        new vscode.Diagnostic(
+                            new vscode.Range(0, 0, 0, 0),
+                            `${finding.findingType}: ${finding.message} - Resource name: ${finding.resourceName}, Policy name: ${finding.policyName} - ${reason.description}`,
+                            vscode.DiagnosticSeverity.Error
+                        )
+                    )
+                    customPolicyCheckDiagnosticCollection.set(IamPolicyChecksWebview.editedDocumentUri, diagnostics)
+                })
+            })
+            jsonOutput.NonBlockingFindings.forEach((finding: any) => {
+                finding.details.reasons.forEach((reason: any) => {
+                    diagnostics.push(
+                        new vscode.Diagnostic(
+                            new vscode.Range(0, 0, 0, 0),
+                            `${finding.findingType}: ${finding.message} - Resource name: ${finding.resourceName}, Policy name: ${finding.policyName} - ${reason.description}`,
+                            vscode.DiagnosticSeverity.Warning
+                        )
+                    )
+                    customPolicyCheckDiagnosticCollection.set(IamPolicyChecksWebview.editedDocumentUri, diagnostics)
+                })
+            })
+            this.onCustomPolicyCheckResponse.fire([
+                'Result: FAIL. Please view the problems panel to review the issues with your policy document. Policy checks should be run until issues are no longer found in your policy document.',
```

Comment:
lets reuse these messages as const

Reply:
Adding constants in future. Adding to ticket.